diff --git a/addons/point_of_sale/static/src/js/db.js b/addons/point_of_sale/static/src/js/db.js
index c2b3fad..abb64c3 100644
--- a/addons/point_of_sale/static/src/js/db.js
+++ b/addons/point_of_sale/static/src/js/db.js
@@ -155,6 +155,9 @@ var PosDB = core.Class.extend({
         if (product.barcode) {
             str += '|' + product.barcode;
         }
+        if (product.second_barcode) {
+            str += '|' + product.second_barcode;
+        }
         if (product.default_code) {
             str += '|' + product.default_code;
         }
@@ -206,6 +209,9 @@ var PosDB = core.Class.extend({
             if(product.barcode){
                 this.product_by_barcode[product.barcode] = product;
             }
+            if(product.second_barcode){
+                this.product_by_barcode[product.second_barcode] = product;
+            }
         }
     },
     _partner_search_string: function(partner){
diff --git a/addons/point_of_sale/static/src/js/models.js b/addons/point_of_sale/static/src/js/models.js
index 84f033b..f4abc8d 100644
--- a/addons/point_of_sale/static/src/js/models.js
+++ b/addons/point_of_sale/static/src/js/models.js
@@ -307,7 +307,7 @@ exports.PosModel = Backbone.Model.extend({
         },
     },{
         model:  'product.product',
-        fields: ['display_name', 'list_price','price','pos_categ_id', 'taxes_id', 'barcode', 'default_code', 
+        fields: ['display_name', 'list_price','price','pos_categ_id', 'taxes_id', 'barcode', 'second_barcode', 'default_code', 
                  'to_weight', 'uom_id', 'description_sale', 'description',
                  'product_tmpl_id','tracking'],
         order:  ['sequence','default_code','name'],
