diff --git a/addons/stock_account/models/product.py b/addons/stock_account/models/product.py
index 0622c16d2b5..c078ac54324 100644
--- a/addons/stock_account/models/product.py
+++ b/addons/stock_account/models/product.py
@@ -239,7 +239,7 @@ class ProductProduct(models.Model):
 
         for product in self:
             if product.cost_method in ['standard', 'average']:
-                qty_available = product.with_context(company_owned=True, owner_id=False).qty_available
+                qty_available = product.with_context(owner_id=False).qty_available
                 price_used = product.standard_price
                 if to_date:
                     price_used = product.get_history_price(
@@ -252,7 +252,7 @@ class ProductProduct(models.Model):
                 if to_date:
                     if product.product_tmpl_id.valuation == 'manual_periodic':
                         product.stock_value = product_values[product.id]
-                        product.qty_at_date = product.with_context(company_owned=True, owner_id=False).qty_available
+                        product.qty_at_date = product.with_context(owner_id=False).qty_available
                         product.stock_fifo_manual_move_ids = StockMove.browse(product_move_ids[product.id])
                     elif product.product_tmpl_id.valuation == 'real_time':
                         valuation_account_id = product.categ_id.property_stock_valuation_account_id.id
