diff --git a/addons/point_of_sale/static/src/js/models.js b/addons/point_of_sale/static/src/js/models.js
index 86a7b44bcfb..06b46ba5645 100644
--- a/addons/point_of_sale/static/src/js/models.js
+++ b/addons/point_of_sale/static/src/js/models.js
@@ -514,16 +514,7 @@ exports.PosModel = Backbone.Model.extend({
         fields: ['name', 'is_cash_count', 'use_payment_terminal'],
         domain: function(self){return ['|',['active', '=', false], ['active', '=', true]]; },
         loaded: function(self, payment_methods) {
-            self.payment_methods = payment_methods.sort(function(a,b){
-                // prefer cash payment_method to be first in the list
-                if (a.is_cash_count && !b.is_cash_count) {
-                    return -1;
-                } else if (!a.is_cash_count && b.is_cash_count) {
-                    return 1;
-                } else {
-                    return a.id - b.id;
-                }
-            });
+            self.payment_methods = payment_methods;
             self.payment_methods_by_id = {};
             _.each(self.payment_methods, function(payment_method) {
                 self.payment_methods_by_id[payment_method.id] = payment_method;
