diff --git a/addons/stock/procurement.py b/addons/stock/procurement.py
index 7b067ed..98857e6 100644
--- a/addons/stock/procurement.py
+++ b/addons/stock/procurement.py
@@ -342,6 +342,7 @@ class procurement_order(osv.osv):
 
         procurement_obj = self.pool.get('procurement.order')
         dom = company_id and [('company_id', '=', company_id)] or []
+        dom.append(('suggest', '=', False))
         orderpoint_ids = orderpoint_obj.search(cr, uid, dom)
         prev_ids = []
         while orderpoint_ids:
