diff --git a/addons/account/__init__.py b/addons/account/__init__.py
index 138004b0849..07e6475f760 100644
--- a/addons/account/__init__.py
+++ b/addons/account/__init__.py
@@ -45,7 +45,9 @@ def _auto_install_l10n(env):
             module_list.append('l10n_de_skr03')
             module_list.append('l10n_de_skr04')
         else:
-            if env['ir.module.module'].search([('name', '=', 'l10n_' + country_code.lower())]):
+            if env['ir.module.module'].search([('name', '=', 'l10n_%s_oca' % country_code.lower())]):
+                module_list.append('l10n_%s_oca' % country_code.lower())
+            elif env['ir.module.module'].search([('name', '=', 'l10n_' + country_code.lower())]):
                 module_list.append('l10n_' + country_code.lower())
             else:
                 module_list.append('l10n_generic_coa')
